File: OpenWeather/Utils/ViewModelLocator.vb
Imports Microsoft.Practices.Unity

Public Class ViewModelLocator
    Private _container As UnityContainer

    Public Sub New()
        _container = New UnityContainer
        _container.RegisterType(Of IWeatherService, OpenWeatherMapService)()
        _container.RegisterType(Of IDialogService, DialogService)()
    End Sub

    Public ReadOnly Property WeatherVM As WeatherViewModel
        Get
            Return _container.Resolve(Of WeatherViewModel)()
        End Get
    End Property
End Class

File: OpenWeatherCS/Utils/ViewModelLocator.cs
using Microsoft.Practices.Unity;
using OpenWeatherCS.Services;
using OpenWeatherCS.ViewModels;

namespace OpenWeatherCS.Utils
{
    public class ViewModelLocator
    {
        private UnityContainer _container;
        public ViewModelLocator()
        {
            _container = new UnityContainer();
            _container.RegisterType<IWeatherService, OpenWeatherMapService>();
            _container.RegisterType<IDialogService, DialogService>();
        }

        public WeatherViewModel WeatherVM
        {
            get { return _container.Resolve<WeatherViewModel>(); }
        }
    }
}
